crates/dojo-lang/src/contract.rs

```diff
@@ -48,29 +48,38 @@ impl DojoContract {
                     use dojo::world;
                     use dojo::world::IWorldDispatcher;
                     use dojo::world::IWorldDispatcherTrait;
+                    use dojo::world::IWorldProvider;
+                   
+                    use dojo::components::upgradeable::upgradeable as upgradeable_component;
 
+                    component!(path: upgradeable_component, storage: upgradeable, event: \
+                 UpgradeableEvent);
+
+                    
                     #[storage]
                     struct Storage {
                         world_dispatcher: IWorldDispatcher,
+                        #[substorage(v0)]
+                        upgradeable: upgradeable_component::Storage,
```

Comment:
better to fully qualify here and avoid import conflicts

```suggestion
                        upgradeable: dojo::components::upgradeable::upgradeable::Storage,
```